react/internal/http-switch-link.js

```diff
@@ -7,42 +7,51 @@
  *
  */
 import {ApolloLink} from 'apollo-link'
-import { prop, map, contains, isEmpty, isNil } from 'ramda'
-import { parse, stringify } from 'querystring'
-import { print } from 'graphql'
+import {parse, format} from 'url'
+import {print} from 'graphql'
+
+const isEmpty = (val) => (val === undefined || val === "" || val === {} || val === [])
+const isNil = (val) => (val === undefined || val === null)
 
 function removeUnusedFields(obj) {
```

Comment:
This same function is in three code bases now (render-server, graphql-server and render-runtime), is it really necessary? Optimizations are only worth it if they cover cases that happen a lot. Can you explain the use case you're trying to optimize for here?